monix-catnap/src/main/scala/com/avast/sst/monix/catnap/CircuitBreakerModule.scala

```diff
@@ -0,0 +1,31 @@
+package com.avast.sst.monix.catnap
+
+import cats.effect.{Clock, Sync}
+import monix.catnap.CircuitBreaker
+
+class CircuitBreakerModule[F[_]](implicit F: Sync[F]) {
+
+  /** Makes [[monix.catnap.CircuitBreaker]] initialized with the given config and [[cats.effect.Clock]]. */
+  def make(config: CircuitBreakerConfig,
+           clock: Clock[F],
+           onRejected: F[Unit] = F.unit,
+           onClosed: F[Unit] = F.unit,
+           onHalfOpen: F[Unit] = F.unit,
+           onOpen: F[Unit] = F.unit): F[CircuitBreaker[F]] = {
+    CircuitBreaker[F].of(config.maxFailures,
+                         config.resetTimeout,
+                         config.exponentialBackoffFactor,
+                         config.maxResetTimeout,
+                         onRejected,
+                         onClosed,
+                         onHalfOpen,
+                         onOpen)(clock)
+  }
+
+}
+
+object CircuitBreakerModule {
+
+  def apply[F[_]: Sync] = new CircuitBreakerModule[F]
+
+}
```

Comment:
why not just 
```scala
object CircuitBreakerModule {
  def make[F[_]: Sync](config: CircuitBreakerConfig,
           clock: Clock[F],
           onRejected: F[Unit] = F.unit,
           onClosed: F[Unit] = F.unit,
           onHalfOpen: F[Unit] = F.unit,
           onOpen: F[Unit] = F.unit): F[CircuitBreaker[F]] = {
    CircuitBreaker[F].of(config.maxFailures,
                         config.resetTimeout,
                         config.exponentialBackoffFactor,
                         config.maxResetTimeout,
                         onRejected,
                         onClosed,
                         onHalfOpen,
                         onOpen)(clock)
  }
}
```
?

Reply:
sorry, I still don't understand. You don't need partial application, because you have just 1 type parameter: `F`, and you _do_ have access to it.

Reply:
I have access to the type parameter but I don't have access to the instance of the type parameter which I need to provide the default values `F.unit`.

Reply:
aaaah, now I see :+1: